### DISCUSSION TITLE
Extra tag or mark in single page

### DISCUSSION THREAD

**Discussion by un4gt** [sphinx-doc/sphinx#13033] [Q&A]
Is it possible to add additional tags to a page and get the added tags in the page through the event handler in the `html-page-context` event?
I want to check if some pages have the tags I added, and if so add some javascript code . Like below :

```python
def insert_javascript_on_marked_page(app, page_name, template_name, context, doctree):
    # TODO

def setup(app):
    app.connect('some_event', insert_javascript_on_marked_page)
```

**Comment by SilverRainZ** [comment]
You can add custom [File-wide metadata](https://www.sphinx-doc.org/en/master/usage/restructuredtext/field-lists.html#file-wide-metadata) to the page, and access them through `app.env.metadata[docname]`.
